lib/std/array_hash_map.zig

```diff
@@ -137,6 +137,23 @@ pub fn ArrayHashMap(
             self.* = undefined;
         }
 
+        /// Puts the hash map into a state where any method call that would
+        /// cause an existing key or value pointer to become invalidated will
+        /// instead trigger an assertion.
+        ///
+        /// An additional call to `lockMutation` in such state also triggers an
+        /// assertion.
+        ///
+        /// `unlockMutation` returns the hash map to the previous state.
+        pub fn lockMutation(self: *Self) void {
```

Comment:
Would `lockRealloc()` be a better name for this? Some mutation is allowed, for example the `fooAssumeCapacity()` function variants as you mentioned.

Reply:
That's a nice name. I'll make that change.

Reply:
On second thought, "realloc" is not the right concept either since it also gets triggered for clearing and shrinking (with retained capacity).

Reply:
Indeed! I think what you've landed on with `lockPointers()` is great :)